packages/launchpad/cypress/e2e/integration/choose-a-browser.spec.ts

```diff
@@ -0,0 +1,273 @@
+import type Sinon from '@packages/../cli/types/sinon'
+import type { FoundBrowser } from '@packages/types/src'
+
+describe('Choose a Browser Page', () => {
+  // Walks through setup pages to get to browser selection
+  const stepThroughConfigPages = () => {
+    cy.get('h1').should('contain', 'Configuration Files')
+    cy.contains('Continue').click()
+    cy.get('h1').should('contain', 'Initializing Config...')
+    cy.contains('Next Step').click()
+  }
+
+  // Mocks server's browser detection with known list
+  const setupMockBrowsers = function () {
+    cy.withCtx(async (ctx, o) => {
+      let mockedGetBrowsers = ctx._apis.appApi.getBrowsers as Sinon.SinonStub
+
+      if (mockedGetBrowsers.restore) {
+        mockedGetBrowsers.restore()
```

Comment:
I thought our framework reset signons automatically. Wondering if this should be sinon.restore() rather than sinon.reset():

https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/cypress/e2e/e2ePluginSetup.ts#L128

Reply:
I think it needs both, now that look at it. Restore will reset mocks to their original implementation, reset will clean up any stand-alone stubs.